### ISSUE TITLE
Mango::Bulk seems to handle errors in a wrong way

### ISSUE THREAD

**Issue by ufobat** [oliwer/mango#14]
I got following stacktrace using Mango::Bulk:

```
Use of uninitialized value in addition (+) at /opt/perl/lib/site_perl/5.20.0/Mango/Bulk.pm line 72.
        Mango::Bulk::_merge("update", 0, HASH(0xbf74600), undef) called at /opt/perl/lib/site_perl/5.20.0/Mango/Bulk.pm line 102
        Mango::Bulk::__ANON__(Mango::Database=HASH(0xbf9d098), "Connect timeout", undef) called at /opt/perl/lib/site_perl/5.20.0/Mango/Database.pm line 60
        Mango::Database::__ANON__(Mango=HASH(0xbfa0b48), "Connect timeout", undef) called at /opt/perl/lib/site_perl/5.20.0/Mango.pm line 202
        Mango::_finish(Mango=HASH(0xbfa0b48), undef, CODE(0xbbff358), "Connect timeout") called at /opt/perl/lib/site_perl/5.20.0/Mango.pm line 174
        Mango::_error(Mango=HASH(0xbfa0b48), "51fa0576f12ccf67ccd20e14064655ac", "Connect timeout") called at /opt/perl/lib/site_perl/5.20.0/Mango.pm line 145
        Mango::__ANON__(Mojo::IOLoop=HASH(0x8745ac8), "Connect timeout", undef) called at /opt/perl/lib/site_perl/5.20.0/Mojo/IOLoop.pm line 65
        Mojo::IOLoop::__ANON__(Mojo::IOLoop::Client=HASH(0xc020518)) called at /opt/perl/lib/site_perl/5.20.0/Mojo/EventEmitter.pm line 15
        Mojo::EventEmitter::emit(Mojo::IOLoop::Client=HASH(0xc020518), "error", "Connect timeout") called at /opt/perl/lib/site_perl/5.20.0/Mojo/IOLoop/Client.pm line 41
        Mojo::IOLoop::Client::__ANON__(Mojo::Reactor::EV=HASH(0x8745eb8)) called at /opt/perl/lib/site_perl/5.20.0/Mojo/Reactor/Poll.pm line 143
        eval {...} called at /opt/perl/lib/site_perl/5.20.0/Mojo/Reactor/Poll.pm line 143
        Mojo::Reactor::Poll::_try(Mojo::Reactor::EV=HASH(0x8745eb8), "Timer", CODE(0xc037a30)) called at /opt/perl/lib/site_perl/5.20.0/Mojo/Reactor/EV.pm line 66
        Mojo::Reactor::EV::__ANON__(EV::Timer=SCALAR(0xc040480), 256) called at /opt/perl/lib/site_perl/5.20.0/x86_64-linux-thread-multi/AnyEvent/Impl/EV.pm line 88

```

**Comment by oliwer** [comment]
After a bit of digging this does not seem to be a protocol issue. You got this error because the program did not receive any response from MongoDB, thus the "connect timeout" in the traces. I suppose this is related to the "localhost" issue you told me about?

**Comment by ufobat** [comment]
Yes it is related to the "Connect Timeout" issue, we talked about. I just assumed that this is wrong error handling, isnt it? Because the Connect Timeout caused another runtime issue.

**Comment by oliwer** [comment]
This should be fixed in version 1.27. Thanks for the report!
